src/middlewared/middlewared/plugins/vmware.py

```diff
@@ -392,6 +392,11 @@ def _dataset_get_vms(self, dataset, recursive):
             ]
         return self.middleware.call_sync("vmware.query", [f])
 
+    @private
+    def snapshot_begin(self, dataset, recursive):
+        qs = self._dataset_get_vms(dataset, recursive)
+        return self.snapshot_proceed(dataset, qs)
+
```

Comment:
Q: in the original commit, `snapshot_begin()` only called `_dataset_get_vms()`; now we _*also*_ trigger `snapshot_proceed()` -- I assume this was intentional or did the underlying functionality change, so that now `snapshot_proceed()` should be called in this context ?

Reply:
@rick-mesta actual `snapshot_begin` was `_dataset_get_vms` + all the body of `snapshot_proceed`, and `snapshot_proceed` did not exist. This type of diff looks confusing sometimes.

Reply:
Thx for the explanation @themylogin ... just wanted to make sure it wasn't an easy oversight 👍🏼